src/server.c

```diff
@@ -133,23 +157,48 @@ void serverLogRaw(int level, const char *msg) {
     } else {
         int off;
         struct timeval tv;
-        int role_char;
         pid_t pid = getpid();
         int daylight_active = atomic_load_explicit(&server.daylight_active, memory_order_relaxed);
 
         gettimeofday(&tv, NULL);
         struct tm tm;
         nolocks_localtime(&tm, tv.tv_sec, server.timezone, daylight_active);
-        off = strftime(buf, sizeof(buf), "%d %b %Y %H:%M:%S.", &tm);
-        snprintf(buf + off, sizeof(buf) - off, "%03d", (int)tv.tv_usec / 1000);
+        switch (server.log_timestamp_format) {
+        case LOG_TIMESTAMP_DEFAULT:
+            off = strftime(buf, sizeof(buf), "%d %b %Y %H:%M:%S.", &tm);
+            snprintf(buf + off, sizeof(buf) - off, "%03d", (int)tv.tv_usec / 1000);
+            break;
+
+        case LOG_TIMESTAMP_ISO8601:
+            off = strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.", &tm);
+            char tzbuf[7];
+            format_timezone(tzbuf, server.timezone, server.daylight_active);
+            snprintf(buf + off, sizeof(buf) - off, "%03d%s", (int)tv.tv_usec / 1000, tzbuf);
+            break;
+
+        case LOG_TIMESTAMP_UNIX: snprintf(buf, sizeof(buf), "%ld", tv.tv_sec); break;
+        default: break;
+        }
+        int role_index;
         if (server.sentinel_mode) {
-            role_char = 'X'; /* Sentinel. */
+            role_index = 0; /* Sentinel. */
         } else if (pid != server.pid) {
-            role_char = 'C'; /* RDB / AOF writing child. */
+            role_index = 1; /* RDB / AOF writing child. */
         } else {
-            role_char = (server.primary_host ? 'S' : 'M'); /* replica or Primary. */
+            role_index = (server.primary_host ? 2 : 3); /* Slave or Master. */
+        }
+        switch (server.log_format) {
+        case LOG_FORMAT_LOGFMT: {
+            fprintf(fp, "pid=%d role=%s timestamp=\"%s\" level=%s message=\"%s\"\n", (int)getpid(), roles[role_index],
+                    buf, verbose_level[level], msg);
+            break;
+        }
+
+        default: {
```

Comment:
```suggestion
        case LOG_TIMESTAMP_ISO8601:
        case LOG_TIMESTAMP_UNIX: {
```
I think this is a bit clearer from a wording perspective.

Reply:
I use `case LOG_FORMAT_DEFAULT ` instead. 

'''
typedef enum { LOG_FORMAT_DEFAULT = 0, LOG_FORMAT_LOGFMT } log_format_type;
'''